qiime2/sdk/action.py

```diff
@@ -394,6 +395,7 @@ def _callable_executor_(self, callable, view_args, output_types,
                 raise TypeError(
                     "Visualizer %r should not return anything. "
                     "Received %r as a return value." % (self, ret_val))
+            provenance.output_name = 'visualization'
```

Comment:
I'm not sold on this here. Using `fork()` when there's only one result feels silly, but so does this a little bit.

Would assignment of `provenance.fork(name)` to `provenance` then using `provenance.output_name` be a better move for the Method version of this? Then they would match.

Reply:
I think it is clear as implemented what is going on here. That said, I think would be fine with either approach.